x-pack/plugins/security_solution/public/timelines/components/timeline/unified_components/data_table/index.tsx

```diff
@@ -0,0 +1,495 @@
+      [timelineId, updatedAt]
+    );
+
+    const cellActionsMetadata = useMemo(() => ({ scopeId: timelineId }), [timelineId]);
+
+    const onUpdateSampleSize = useCallback(
+      (newSampleSize: number) => {
+        if (newSampleSize !== sampleSize) {
+          dispatch(timelineActions.updateSampleSize({ id: timelineId, sampleSize: newSampleSize }));
+          refetch();
+        }
+      },
+      [dispatch, sampleSize, timelineId, refetch]
+    );
+
+    const onUpdateRowHeight = useCallback(
+      (newRowHeight: number) => {
+        if (newRowHeight !== rowHeight) {
+          dispatch(timelineActions.updateRowHeight({ id: timelineId, rowHeight: newRowHeight }));
+        }
+      },
+      [dispatch, rowHeight, timelineId]
+    );
+
+    const services = useMemo(() => {
+      return {
+        theme,
+        fieldFormats,
+        storage,
+        toastNotifications: toastsService,
+        uiSettings,
+        dataViewFieldEditor,
+        data: dataPluginContract,
+      };
+    }, [
+      theme,
+      fieldFormats,
+      storage,
+      toastsService,
+      uiSettings,
+      dataViewFieldEditor,
+      dataPluginContract,
+    ]);
+
+    if (!dataView) {
+      return null;
+    }
+
+    return (
+      <StatefulEventContext.Provider value={activeStatefulEventContext}>
+        <StyledTimelineUnifiedDataTable>
+          {(dataLoadingState === DataLoadingState.loading ||
+            dataLoadingState === DataLoadingState.loadingMore) && (
+            <StyledEuiProgress data-test-subj="discoverDataGridUpdating" size="xs" color="accent" />
+          )}
+          <DataGridMemoized
+            ariaLabelledBy="timelineDocumentsAriaLabel"
+            className={'udtTimeline'}
+            columns={defaultColumnIds}
+            expandedDoc={expandedDoc}
+            dataView={dataView}
+            showColumnTokens={true}
+            loadingState={dataLoadingState}
+            onFilter={onAddFilter as DocViewFilterFn}
+            onResize={onResizeDataGrid}
+            onSetColumns={onSetColumns}
+            onSort={!isTextBasedQuery ? onSort : undefined}
+            rows={tableRows}
+            sampleSizeState={sampleSize || 500}
+            onUpdateSampleSize={onUpdateSampleSize}
+            setExpandedDoc={onSetExpandedDoc}
+            settings={tableSettings}
+            showTimeCol={showTimeCol}
+            isSortEnabled={true}
+            sort={sortingColumns}
+            rowHeightState={3}
+            isPlainRecord={isTextBasedQuery}
+            rowsPerPageState={itemsPerPage}
+            onUpdateRowsPerPage={onChangeItemsPerPage}
+            onUpdateRowHeight={onUpdateRowHeight}
+            onFieldEdited={onFieldEdited}
+            cellActionsTriggerId={SecurityCellActionsTrigger.DEFAULT}
+            services={services}
+            visibleCellActions={3}
+            externalCustomRenderers={customColumnRenderers}
+            renderDocumentView={() => <></>}
```

Comment:
please pull the fn out of this component pass it through :) . I'm pretty sure there's prob a `noop` fn you can import

Reply:
It was done as well.